.eslintrc.js

```diff
@@ -1,3 +1,5 @@
 const neutrino = require('./packages/neutrino');
 
+process.env.NODE_ENV = process.env.NODE_ENV || 'lint';
```

Comment:
Possible values instead of "lint" might be "test" or "production". Though given we use "test" for `karma.config.js` etc, it seems wrong to just call this "production". 

ESLint config conditionals that choose to rely on `NODE_ENV` (eg enabling/disabling the `no-console` rule) should be using `NODE_ENV !== 'development'` rather than `NODE_ENV === 'production'` for compatibility with `NODE_ENV='test'` anyway -- so using `NODE_ENV='lint'` doesn't seem like it would cause any more problems.

Reply:
@eliperelman thoughts about this part?

Reply:
I'm OK with this, although my initial gut reaction wishes this existed in `eslint()` instead of being dotfile boilerplate.

Reply:
Yeah me too -- though I guess that would mean we would have to change the whole way the middlewares are instantiated.

Reply:
Yeah which is too much right now. The thing that prevents us from doing this is not being able to reliably infer when the `NODE_ENV` was set by Neutrino vs. the environment.

Reply:
That definitely gets us closer, however if I follow correctly, it would mean that the web preset (and any others) see the NODE_ENV as `'production'` and not `'test'` -- and that even for `.karma()`, parts of the config still comes from other presets.

Therefore it seems that the only way to solve this is make the meat of the presets be a step that occurs **after** they register themselves.

ie:
```js
// registers the presets and what output commands they include, but doesn't actually generate any config or adjust NODE_ENV
neutrino.use(some_lint_preset);
neutrino.use(some_build_preset);
neutrino.use(some_test_preset);

// The output function then sets NODE_ENV if undefined (in this case to `"test"`)
// and then actually generates the config, where all presets get to see `NODE_ENV=test`
neutrino.karma();
```

Reply:
(but that's a major change and would mean backwards incompatibility with all existing presets)

Reply:
I think keeping middleware compat is more important here.